nbodykit/algorithms/fof.py

```diff
@@ -63,7 +63,11 @@ def __init__(self, source, linking_length, nmin, absolute=False, periodic=True,
 
         # linking length relative to mean separation
         if not absolute:
-            mean_separation = pow(numpy.prod(source.attrs['BoxSize']) / source.csize, 1.0 / len(source.attrs['Nmesh']))
+            try:
+                ndim = len(source.attrs['Nmesh'])
+            except KeyError:
+                ndim = source['Position'].shape[1]
```

Comment:
What about avoid reraise with:
  if 'Nmesh' not in source.attrs

Reply:
Please look at the new changes.

And what do you mean by to "avoid reraise"? Thanks!

Reply:
Not exactly a reraise -- but if there is an exception in line 69, it would be exception inside exception handler.